app/models/communitarian/resolution.rb

```diff
@@ -15,25 +13,23 @@ def initialize(resolution_schedule)
     end
 
     def reopen_weekly_resolution!(original_post)
-      return if to_be_closed?(original_post)
+      return if to_be_closed?(original_post) || has_recent_polls?(original_post.topic_id)
 
       post_attributes = original_post.attributes.slice(*self.class::REOPENED_RESOLUTION_ATTRIBUTES)
       post = Post.create!(post_attributes)
       poll = post.reload.polls.first
-      poll.update!(close_at: resolution_schedule.next_close_time(poll.close_at))
+      poll.update!(close_at: resolution_schedule.next_close_time)
       self.schedule_jobs(post)
     end
 
     def schedule_jobs(post)
-      return if to_be_closed?(post)
-
       ::DiscoursePoll::Poll.schedule_jobs(post)
+      return if to_be_closed?(post)
 
-      poll = post.reload.polls.first
       job_args = { post_id: post.id }
-
       Jobs.cancel_scheduled_job(:reopen_resolution, job_args)
-      Jobs.enqueue_at(resolution_schedule.reopen_delay.since(poll.close_at), :reopen_resolution, job_args)
+      Jobs.enqueue_at(1.minute.from_now, :open_poll, job_args.merge(poll_name: "poll"))
```

Comment:
I've added this job, because after post creation `:close_poll` job can run before `poll.close_at` updated and poll will be closed immediately.